### ISSUE TITLE
Create VPN connection between MP and Azure FNG

### ISSUE THREAD

**Issue by dms1981** [ministryofjustice/modernisation-platform#2739]
## User Story

As a migration team member
I want a VPN connection between my Azure environment and the Modernisation Platform
So that I have a performant network path within my administrative control to use when migrating data 

## User Type(s)

Modernisation Platform Engineer - will configure the VPN and routing on the AWS side
Modernisation Platform Customer - will configure the VPN and routing on the Azure side

## Value

Migration teams have experienced issues in moving data from their Azure environments to the replatformed applications on the Modernisation Platform. By creating a VPN directly between the MP Transit Gateway and the Azure Fix n' Go environment we will give our customers a network path that is within their administrative control with sufficient performance for their needs.

## Questions / Assumptions / Hypothesis

The DSO team will commence the work by creating an Azure VPN Gateway and supplying us with details so that we can configure the VPN between our Transit Gateway and Azure

Traffic between MP and Azure FNG will be routed via our Network Firewall

The VPN will be tactical in nature with an anticipated decommissioning date of July 2023

Routing will be static; while BGP is an option, it would expand the scope of this story and require close coordination between DSO and MP

## Definition of done

- [ ] VPN tunnel established between environments
- [ ] Static routes entered on both sides of the VPN
- [ ] Necessary firewall rules supplied by DSO team

## Reference

[How to write good user stories](https://www.gov.uk/service-manual/agile-delivery/writing-user-stories)

**Comment by dms1981** [comment]
This thread contains some contextual discussion: https://mojdt.slack.com/archives/C013RM6MFFW/p1669887678555989

**Comment by dms1981** [comment]
Created VPN through https://github.com/ministryofjustice/modernisation-platform/pull/2743 and passed config across to DSO team.
Will need them to complete their work before we can close this one successfully. It's possible that we'll need another PR to add some options to our VPN tunnel creation code, and definite that we'll need a PR to add routes via the new VPN

**Comment by dms1981** [comment]
VPN tunnels have been successfully established and we are advertising routes to Azure.
However, given the large number of routes that we receive back from Azure we can't use route propagation without complicating our route tables. I'll need to raise a PR to add static routes via the VPN connections.

**Comment by dms1981** [comment]
Two tunnels created, routes advertised into Azure, with static routes to Azure destinations now in relevant MP TGW route tables

**Comment by dms1981** [comment]
https://github.com/ministryofjustice/modernisation-platform/pull/2735
https://github.com/ministryofjustice/modernisation-platform/pull/2743
https://github.com/ministryofjustice/modernisation-platform/pull/2783
https://github.com/ministryofjustice/modernisation-platform/pull/2789
https://github.com/ministryofjustice/modernisation-platform/pull/2794
https://github.com/ministryofjustice/modernisation-platform/pull/2797
